### PR TITLE
[CI][Tests] Add BWC tests for 2.7 and 2.8

### PR DESCRIPTION
**Pull request by manasvinibs**
### Description

Test data generated with:

yarn test:bwc -o https://artifacts.opensearch.org/releases/bundle/opensearch/2.6.0/opensearch-2.6.0-linux-x64.tar.gz -d https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.6.0/opensearch-dashboards-2.6.0-linux-x64.tar.gz -g true

yarn test:bwc -o https://artifacts.opensearch.org/releases/bundle/opensearch/2.7.0/opensearch-2.7.0-linux-x64.tar.gz -d https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.7.0/opensearch-dashboards-2.7.0-linux-x64.tar.gz -g true

yarn test:bwc -o https://artifacts.opensearch.org/releases/bundle/opensearch/2.6.0/opensearch-2.6.0-linux-x64.tar.gz -d https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.6.0/opensearch-dashboards-2.6.0-linux-x64.tar.gz -g true -s true

yarn test:bwc -o https://artifacts.opensearch.org/releases/bundle/opensearch/2.7.0/opensearch-2.7.0-linux-x64.tar.gz -d https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.7.0/opensearch-dashboards-2.7.0-linux-x64.tar.gz -g true -s true

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3776

### Check List

- [X] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ananzh — APPROVED** [review]

**Review by abbyhu2000 — DISMISSED** [review]

**Review by ananzh — DISMISSED** [review]

**Comment by manasvinibs** [comment]
There is a cypress error while generating the data (without-security) for timeline visualization using 2.6 and 2.7 artifacts. 
```
Generating BWC test data without security
       adds Timeline visualization:
     CypressError: Timed out retrying after 60050ms: `cy.click()` failed because this element is `disabled`:

`<button class="euiButton euiButton--primary euiButton--small euiButton-isDisabled" disabled="" type="button" data-test-subj="visualizeSaveButton">...</button>`

Fix this problem, or use `{force: true}` to disable error checking.

https://on.cypress.io/element-cannot-be-interacted-with
      at $Cy.ensureNotDisabled (http://localhost:5601/__cypress/runner/cypress_runner.js:163868:76)
      at runAllChecks (http://localhost:5601/__cypress/runner/cypress_runner.js:150539:14)
      at retryActionability (http://localhost:5601/__cypress/runner/cypress_runner.js:150616:16)
      at tryCatcher (http://localhost:5601/__cypress/runner/cypress_runner.js:13022:23)
      at Function.Promise.attempt.Promise.try (http://localhost:5601/__cypress/runner/cypress_runner.js:10296:29)
      at whenStable (http://localhost:5601/__cypress/runner/cypress_runner.js:168808:63)
      at http://localhost:5601/__cypress/runner/cypress_runner.js:168305:14
      at tryCatcher (http://localhost:5601/__cypress/runner/cypress_runner.js:13022:23)
      at Promise._settlePromiseFromHandler (http://localhost:5601/__cypress/runner/cypress_runner.js:10957:31)
      at Promise._settlePromise (http://localhost:5601/__cypress/runner/cypress_runner.js:11014:18)
      at Promise._settlePromise0 (http://localhost:5601/__cypress/runner/cypress_runner.js:11059:10)
      at Promise._settlePromises (http://localhost:5601/__cypress/runner/cypress_runner.js:11139:18)
      at Promise._fulfill (http://localhost:5601/__cypress/runner/cypress_runner.js:11083:18)
      at http://localhost:5601/__cypress/runner/cypress_runner.js:12697:46
  From Your Spec Code:
      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress/integration/without-security/helpers/generate_data.js:150:54)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        4                                                                                │
  │ Passing:      3                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     1 minute, 27 seconds                                                             │
  │ Spec Ran:     without-security/helpers/generate_data.js                                        │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /home/ubuntu/OpenSearch-Dashboards/bwc_tmp/test/cypress/screenshots/without-secu     (1280x720)
     rity/helpers/generate_data.js/Generating BWC test data without security -- adds                
     Timeline visualization (failed).png                                                         
   ```

**Comment by ananzh** [comment]
@manasvinibs could you check the test failure?

**Review by ananzh — DISMISSED** [review]
LGTM

File: cypress/integration/with-security/helpers/generate_data.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ describe('Generating BWC test data with security', () => {
       });
     // update default expression to use `.es(*)` instead of `.opensearch(*)` for bwc
     cy.get('[class="view-line"]').type('.es(*)');
+    cy.get('[data-test-subj="visualizeEditorRenderButton"]').click();
```
Comment (manasvinibs) [inline cypress/integration/with-security/helpers/generate_data.js:87]: In this PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2720 we recently introduced a change where the default expression for timeline changed from .es(*) to .opensearch(*) , so now we have one more step in the workflow where we need to click `update` button when we create timeline vis before we `save`.  So this line is to get the update button click through.

**Review by ananzh — APPROVED** [review]

**Review by Flyingliuhub — APPROVED** [review]

**Comment by ananzh** [comment]
@manasvinibs I think bwc tests still fail due to `Unable to find an artifact with the name: linux-x64-3.0.0` as shown here https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/5160037751/jobs/9410249664

@kavilla mentioned that bwc test utilizes the build output artifact from build-min-artifact-tests task and that artifact only gets hosted by github for 1 day (intentionally). So this PR should be re-checked after we have a PR to fix it. 

If we need to merge it early, I am okay with manual tests and screenshots posted here.

File: cypress/integration/with-security/helpers/generate_data.js (hunk 1 above)
Comment (joshuarrrr) [reply cypress/integration/with-security/helpers/generate_data.js:87]: sorry I missed this! thanks for the update!

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**